src/set_partner_key.c

```diff
@@ -73,10 +73,16 @@ int set_partner_key(const command_t *cmd) {
         return reply_error(err);
     }
 
-    // The incoming partner name is NOT NULL terminated, so we use memcpy and
-    // manually NULL terminate the buffer
-    memset(G_swap_ctx.partner.name, 0, sizeof(G_swap_ctx.partner.name));
-    memcpy(G_swap_ctx.partner.name, partner.bytes, partner.size);
+    memset(G_swap_ctx.partner.prefixed_name, 0, sizeof(G_swap_ctx.partner.prefixed_name));
+    // Prepare the prefix for FUND display
+    memcpy(&G_swap_ctx.partner.prefixed_name,
+           PARTNER_NAME_PREFIX_FOR_FUND,
+           sizeof(PARTNER_NAME_PREFIX_FOR_FUND));
+    // The incoming partner name is NOT NULL terminated, so we use memcpy.
+    // Don't erase the prefix copied above
+    memcpy(G_swap_ctx.partner.prefixed_name + sizeof(PARTNER_NAME_PREFIX_FOR_FUND) - 1,
+           partner.bytes,
+           partner.size);
```

Comment:
I think you could safely use `strlcpy` and `strlcat` or maybe `snprintf` to make the construct more obvious but this is equivalent.